BrainPortal/app/controllers/tasks_controller.rb

```diff
@@ -208,7 +208,8 @@ def new #:nodoc:
 
     # Filter list of files as provided by the get request
     file_ids = params[:file_ids] || []
-    @files   = Userfile.find_accessible_by_user(file_ids, current_user, :access_requested => :write) rescue []
+    access   = @task.class.properties[:readonly_input_files] ? :read : :write
```

Comment:
I don't like the name of the property... :readonly_input_files is not a good descriptor of what it means for the behavior of the task.

I think it shoudl be something like    :does_not_modify_input_files